units/crons.go

```diff
@@ -760,73 +768,15 @@ func PopulateHostCreationJobs(env evergreen.Environment, part int) amboy.QueueOp
 			"message": "uninitialized hosts",
 			"number":  len(hosts),
 			"runner":  "hostinit",
+			"source":  "PopulateHostCreationJobs",
 		})
 
-		runningHosts, err := host.Find(db.Query(host.IsLive()))
-		if err != nil {
-			return errors.Wrap(err, "getting live hosts")
-		}
-		runningDistroCount := make(map[string]int)
-		for _, h := range runningHosts {
-			runningDistroCount[h.Distro.Id] += 1
-		}
-
-		ts := utility.RoundPartOfHour(part).Format(TSFormat)
-		submitted := 0
-
-		// shuffle hosts because hosts will always come off the index in insertion order
-		// and we want all of them to get an equal chance at being created on this pass
-		// (Fisher-Yates shuffle)
-		rand.Seed(time.Now().UnixNano())
-		for i := len(hosts) - 1; i > 0; i-- {
-			j := rand.Intn(i + 1)
-			hosts[i], hosts[j] = hosts[j], hosts[i]
-		}
-
-		// refresh HostInit
-		if err := env.Settings().HostInit.Get(env); err != nil {
-			return errors.Wrap(err, "getting host init config")
-		}
-		throttleCount := 0
 		catcher := grip.NewBasicCatcher()
+		ts := utility.RoundPartOfHour(part).Format(TSFormat)
 		for _, h := range hosts {
-			if !h.IsSubjectToHostCreationThrottle() {
-				// pass:
-				//    always start spawn hosts asap
-
-			} else {
-				num := runningDistroCount[h.Distro.Id]
-				if num == 0 || num < len(runningHosts)/100 {
-					// if there aren't many of these hosts up, start them even
-					// if `submitted` exceeds the throttle, but increment each
-					// time so we only create hosts up to the threshold
-					runningDistroCount[h.Distro.Id] += 1
-				} else {
-					if submitted > env.Settings().HostInit.HostThrottle {
-						// throttle hosts, so that we're starting very
-						// few hosts on every pass. Hostinit runs very
-						// frequently, lets not start too many all at
-						// once.
-						throttleCount++
-						continue
-					}
-				}
-				// only increment for task hosts, since otherwise
-				// spawn hosts and hosts spawned by tasks could
-				// starve task hosts
-				submitted++
-			}
-
 			catcher.Wrapf(queue.Put(ctx, NewHostCreateJob(env, h, ts, 0, false)), "enqueueing host creation job for host '%s'", h.Id)
```

Comment:
Is there a race condition between this and the various places where we create an intent host and then later create a NewHostCreateJob? I think no because of scopes, but wanted to ask.

Reply:
That's my understanding as well: this relies on scopes to prevent multiple jobs for [the same host](https://github.com/evergreen-ci/evergreen/blob/6dace20f6ad21225fe63f04046f7d047d085ea48/units/provisioning_create_host.go#L62). Because `SetEnqueueAllScopes` is [true](https://github.com/evergreen-ci/evergreen/blob/6dace20f6ad21225fe63f04046f7d047d085ea48/units/provisioning_create_host.go#L63) whichever comes second will fail to enqueue a job.